### PR TITLE
Fix additionalEditorconfig property not being accounted for up-to-date checks and caching

### LINKED ISSUE #750: Caches work incorrectly: additionalEditorconfig should be @Input instead of @Internal
Steps to reproduce:

1. Write some code with ktlint violations
2. Run ktlint via plugin with enabled build cache
3. Disable the violated rule using `additionalEditorconfig`
4. Run ktlint again with enabled build cache

Expected behaviour: ktlint will be re-run with updated editorconfig and will not report the violations

Actual behaviour: ktlint will use the cached result

As far as I understand, the cause of this is that `BaseKtLintCheckTask.additionalEditorconfig` is not checked when computing cache keys for the task because it is marked as `@Internal`, and should be `@Input` instead.

Unless there are any reasons for this property being `@Internal`, I would gladly submit a PR changing it to `@Input`.

### PR DESCRIPTION
**Pull request by m-burst**
Fixes #750

### CODE REVIEW

File: plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/EditorConfigTests.kt [hunk 1]
```diff
@@ -147,6 +147,26 @@ class EditorConfigTests : AbstractPluginTest() {
         }
     }
 
+    @DisplayName("Check task should rerun if additionalEditorconfig property changes")
+    @CommonTest
+    fun checkRerunOnAdditionalEditorconfigPropertyChange(gradleVersion: GradleVersion) {
```
Comment (m-burst) [inline plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/EditorConfigTests.kt:152]: I made sure that the test fails if `val additionalEditorconfig` is `@Internal` before changing it to `@Input`

**Review by wakingrufus — APPROVED** [review]
LGTM Thanks!

**Comment by wakingrufus** [comment]
You will need to update the changelog to pass checks as well

**Comment by m-burst** [comment]
Updated the changelog

**Review by JLLeitschuh — APPROVED** [review]
Thank you for fix, and the unit test, both are greatly appreciated! TDD for the win!

**Pull request merged**